File: javaproject/src/main/java/service/EstatisticaTurnos.java
package service;

import model.*;
import java.util.List;

public class EstatisticaTurnos extends EstatisticaBase {
    private static final String[] TURNOS = { "MATUTINO", "VESPERTINO", "NOTURNO", "INTEGRAL" };

    @Override
    public List<Double> calcular() {
        return calcularPorcentagens(TURNOS, pessoa -> {
            for (Curso curso : cursos) {
                if (curso.candidatos.contains(pessoa)) {
                    return curso.getTurno();
                }
            }
            return "";
        });
    }
}

File: pythonproject/service/estatistica_turnos.py
from service.estatistica_base import EstatisticaBase
from typing import List

class EstatisticaTurnos(EstatisticaBase):
    TURNOS = ["MATUTINO", "VESPERTINO", "NOTURNO", "INTEGRAL"]

    def calcular(self) -> List[float]:
        return self.calcular_porcentagens(self.TURNOS, self.extrair_turno)
    
    def extrair_turno(self, pessoa):
        for curso in self.cursos:
            if pessoa in curso.get_candidatos():
                return curso.get_turno()
        return ""